pkg/endpoint/endpoint.go

```diff
@@ -2359,8 +2363,8 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 
 	// Since we unlocked the endpoint and re-locked, the label update may already be obsolete
 	if e.identityResolutionIsObsolete(myChangeRev) {
-		e.Mutex.RUnlock()
-		elog.Debug("Endpoint identity has changed, aborting resolution routine in favour of new one")
+		defer elog.Debug("Endpoint identity has changed, aborting resolution routine in favour of new one")
```

Comment:
Why the defer? I think it's fine to move it out of the critical section but then let's just swap and not sure defer.

Reply:
Seems like a leftover from when RUnlock returned an error. Done